src/scripts/main.js

```diff
@@ -1,3 +1,355 @@
+
+        setTimeout(() => {
+          cellsInCollumn[j].removeAttribute('scale');
+        }, 200);
+        cellsInCollumn[j + 1].innerText = '';
+        rows[i].append(cellsInCollumn[j + 1]);
+      };
+    };
+  };
+}
+
+function rowsRewriteToRight() {
+  const rows = document.querySelectorAll('tr');
+
+  for (let i = 0; i < [...rows].length; i++) {
+    const cellsInRow = rows[i].children;
+
+    [...cellsInRow].forEach(cell => {
+      if (cell.innerText === '') {
+        cell.parentElement.prepend(cell);
+      }
+    });
+
+    for (let j = cellsInRow.length - 1; j >= 0; j--) {
+      if (cellsInRow[j]
+          && cellsInRow[j - 1]
+          && cellsInRow[j].innerText === cellsInRow[j - 1].innerText
+          && cellsInRow[j].innerText !== '') {
+        cellsInRow[j].innerText = cellsInRow[j].innerText * 2;
+        score += +cellsInRow[j].innerText;
+
+        cellsInRow[j].setAttribute('scale', '1');
+
+        setTimeout(() => {
+          cellsInRow[j].removeAttribute('scale');
+        }, 200);
+        cellsInRow[j - 1].innerText = '';
+        rows[i].prepend(cellsInRow[j - 1]);
+      };
+    };
+  };
+};
+
+function createArrWithAllNumbers(arrForPush) {
+  const verticalTable = document.querySelector('tbody')
+    .classList.contains('tbody-columns');
+
+  if (verticalTable) {
+    const allCells = [...document.querySelectorAll('td')];
+
+    for (let i = 0; i < [...document.querySelectorAll('tr')].length; i++) {
+      const row = allCells.filter(cell => cell.cellIndex === i);
+      const rowNumbers = row.map(cell => cell.innerText);
+
+      arrForPush.push(...rowNumbers);
+    };
+  } else {
+    const allCells = [...findAllCells(document)];
+
+    allCells.map(cell => arrForPush.push(cell.innerText));
+  };
+};
+
+function arrIsChanged(arrBefore, arrAfter) {
+  for (let i = 0; i < arrAfter.length; i++) {
+    if (arrAfter[i] !== arrBefore[i]) {
+      createNumber();
+      break;
+    };
+  };
+};
+
+function insertTable(toTableType, fromTableType) {
+  if (toTableType === 'vetical') {
+    if (fromTableType) {
+      turnTheTable(tbodyVertical, 'column');
+      tbodyHorisontal.remove();
+      table.append(tbodyVertical);
+    };
+  }
+
+  if (toTableType === 'horisontal') {
+    if (!fromTableType) {
+      turnTheTable(tbodyHorisontal, 'row');
+      tbodyVertical.remove();
+      table.append(tbodyHorisontal);
+    };
+  }
+};
+
+function keydownListener(e) {
+  const cellsArr = [];
+
+  createArrWithAllNumbers(cellsArr);
+
+  const horisontalTable = !document.querySelector('tbody')
+    .classList.contains('tbody-columns');
+
+  if (e.key === 'ArrowUp') {
+    insertTable('vetical', horisontalTable);
+    rowsRewriteToLeft();
+  };
+
+  if (e.key === 'ArrowDown') {
+    insertTable('vetical', horisontalTable);
+    rowsRewriteToRight();
+  };
+
+  if (e.key === 'ArrowLeft') {
+    insertTable('horisontal', horisontalTable);
+    rowsRewriteToLeft();
+  };
+
+  if (e.key === 'ArrowRight') {
```

Comment:
It is better to use switch...case instead
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch